File: js/bubble_sort.js
function bubbleSort(array) {
    needsNextPass = true;
    for (let k = 1; k < array.length && needsNextPass; k++) {
        needsNextPass = false;
        for (let i = 0; i < array.length - k; i++) {
            if (array[i] > array[i + 1]) {
                const temp = array[i];
                array[i] = array[i + 1];
                array[i + 1] = temp;
                needsNextPass = true;
            }
        }
    }
}

function testBubbleSort() {
    output.innerHTML += "<h3>testing bubble sort in javascript...</h3>";
    const unsortedArray = getRandArray(10, 1);
    output.innerHTML += `<p>unsorted array -> [${unsortedArray}]</p>`;
    bubbleSort(unsortedArray);
    output.innerHTML += `<p>sorted array -> [${unsortedArray}]</p>`;
}

testBubbleSort();

File: python/bubble_sort.py
def bubble_sort(array):
    needsNextPass = True
    k = 1
    while k < len(array) and needsNextPass:
        needsNextPass = False
        for i in range(len(array) - k):
            if array[i] > array[i + 1]:
                temp = array[i]
                array[i] = array[i + 1]
                array[i + 1] = temp
                needsNextPass = True
        k += 1


print("testing bubble sort in python...")

unsortedArray = [4, 1, 5, 2, 8, 3, 9, 7, 6]
print("unsorted array -> " + str(unsortedArray))
bubble_sort(unsortedArray)
print("sorted array -> " + str(unsortedArray))